processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java

```diff
@@ -268,7 +268,9 @@ private void processResult(List<RawResultIterator> resultIteratorList) throws Ex
    * @return
    */
   private Object[] prepareStreamingRowObjectForSorting(Object[] row) {
-    List<CarbonDimension> dimensions = segmentProperties.getDimensions();
+    List<CarbonDimension> dimensions = new ArrayList<>();
```

Comment:
Don't create a list and add here, every row it creates list object unnecessarily. Please create at the class level and use it.

Reply:
Thanks for your comment @ravipesala, I have a question that are the following two pieces of code equivalent? 
```
List<CarbonDimension> dimensions = new ArrayList<>();
dimensions.addAll(segmentProperties.getDimensions());
dimensions.addAll(segmentProperties.getComplexDimensions());
```
```
carbonTable.getDimensionByTableName(tableName);
```

Reply:
done.